### ISSUE TITLE
Allow JWT header `kid` to identify the credentials in use

### ISSUE THREAD

**Issue by Floby** [Kong/kong#2858]
### Summary

Currently, the JWT plugin looks at a configured claim (`iss` by default). I would like to have the key being used to sign the token to be specified as the [`kid`](https://tools.ietf.org/html/rfc7516#section-4.1.6) header. This seems like a incomplete implementation since this header is specifically designed for this use-case. If this is a feature that has a go, I'm happy to write a PR.

### Steps To Reproduce

1. Create an API
2. Enable the JWT plugin
3. Make a request to this API with a signed JWT with a kid header.

### Additional Details & Logs

- Kong version (`0.11`)
- Kong configuration (registered APIs/Plugins & configuration file)

```yaml
host: 'localhost:8001'
headers: []
apis:
  - name: Admin
    plugins:
      - name: jwt
        attributes:
          enabled: true
          config:
            claims_to_verify: {}
            key_claim_name: kid
            anonymous: ''
            secret_is_base64: false
            uri_param_names: {}
    attributes:
      uris:
        - /admin
      strip_uri: true
      preserve_host: false
      upstream_url: 'http://nginx-lb:8001'
      retries: 5
      upstream_connect_timeout: 60000
      upstream_read_timeout: 60000
      upstream_send_timeout: 60000
      https_only: false
      http_if_terminated: false
consumers:
  - username: Floby's app
    custom_id: floby
    acls: []
    credentials:
      - name: jwt
        attributes:
          algorithm: HS256
          key: floby-sym
          secret: salut
plugins: []
```

**Comment by thibaultcha** [comment]
> If this is a feature that has a go, I'm happy to write a PR.

Happy to receive one!

**Comment by jfalkbjer** [comment]
Any feedback about this feature?
Right now I send the kid both in the JWT header and body. It would be great to be able to remove it from the body.

**Comment by Floby** [comment]
Sorry, I don't have time anymore to look into this :/ but it didn't seem very complicated

**Comment by alexandermontgomery** [comment]
Hello!
I am looking to use Kong's JWT with [Firebase](https://firebase.google.com/docs/auth/admin/verify-id-tokens). Am I right in assuming that Kongs JWT can support multiple public keys, each with its own `config.key_claim_name`? If yes, then it seems like moving the kid to the header would fix this and we can work on PR, but wanted to confirm this is actually the case.

**Comment by thibaultcha** [comment]
This should have been addressed by #3313 - thank you for the suggestion!

**Comment by voroninp** [comment]
Is there any possibility to get plugin version with this fix not upgrading Kong version?

We are sitting on `0.11.1` and switching right now to newer version is painful because there were breaking api changes.
